### ISSUE TITLE
bug: API Gateway incorrectly canoncalizes headers

### ISSUE THREAD

**Issue by mcheng-caring** [localstack/localstack#4532]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

localstack v0.12.17 released with this enhancement that downcases all headers:

> canonicalize HTTP header names in API GW Lambda events to lower-case
> https://github.com/localstack/localstack/commit/657d9fe4bd0e3c4284e6fa67aec62d265d5d9fff

This does not match what we observe in production - custom headers are not downcased. See this part of the docs on API Gateway and how it canonicalizes headers:

> API Gateway enacts the following restrictions and limitations when handling methods with either Lambda integration or HTTP integration.
> * Header names and query parameters are processed in a case-sensitive way.
> * The following table lists the headers that may be dropped, remapped, or otherwise modified when sent to your integration endpoint or sent back by your integration endpoint. In this table: 
> https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-known-issues.html

### Expected Behavior

I would expect that localstack follows the same canonicalization rules that API Gateway implements.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack:0.12.17

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

Used [Insomnia](https://insomnia.rest/) client to issue a POST request to an AWS Lambda function via API Gateway with custom headers. For example, `X-A-Custom-Header`. Attempt to read the custom header in your lambda function case sensitively.

### Environment

```markdown
- OS: macOS for POST request + whatever the docker container uses for localstack
- LocalStack: 0.12.17
```

**Comment by SamMcFaddenBJSS** [comment]
+1 This is causing issues with our build as it is changing our Authorization header to lowercase authorization. This is not the same behaviour as AWS...

**Comment by calvernaz** [comment]
Thanks for reporting @mcheng-caring. This should be fixed in #5213, can you please pull the latest Docker image and give it another try?
